### DISCUSSION TITLE
Please help me to better understand the GPU setup instuctions for OpenWeb UI in conjunction with Ollama

### DISCUSSION THREAD

**Discussion by pascal456** [open-webui/open-webui#6057] [Q&A]
I set up ollama and openweb ui with docker compose. For Ollama I have set the resources [like shown by example in Github](https://github.com/open-webui/open-webui/blob/main/docker-compose.gpu.yaml), which is:

```
deploy:
  resources:
    reservations:
      devices:
        - driver: nvidia
          count: 1
          capabilities: [gpu]
```

GPU support for Ollama works. Now there are instructions for setting up ollama as a separate service and [as a bundled service in conjunction with openweb ui](https://docs.openwebui.com/getting-started#installing-open-webui-with-bundled-ollama-support).

What I do not really understand here is the difference between `open-webui:cuda` and `open-webui:ollama`, both are described to run with `--gpus=all` (docker cli), or basically the code snipped from above in my docker compose scenario.

When ollama already runs in *GPU mode* in a separate service, is there still a point in using the image `open-webui:cuda`? Is this the bundeled version?

Or is it the same as running ollama as a separate service with GPU support and running openweb ui with the `open-webui:main` image?

What confuses me here are the instructions:
- *If Ollama is on a Different Server, use this command: [...]* (using `open-webui:main`)
- *To run Open WebUI with Nvidia GPU support, use this command: [...]* (using `open-webui:cuda`)

and yet different [instructions for bundeled](https://docs.openwebui.com/getting-started#installing-open-webui-with-bundled-ollama-support):

- *With GPU Support: [...]* (using `open-webui:ollama`)

**Note:** I have already [asked this question on Reddit](https://www.reddit.com/r/OpenWebUI/comments/1fq0sad/please_help_me_to_better_understand_the_gpu_setup/) but did not receive any comments that fully answer my question. This is not spam, I just felt like the audience here might be more informed.

**Comment by robw-ddsn** [comment]
Agreed, the instructions are ambiguous and the Reddit answer is unhelpful. I wonder if OpenWebUI can benefit from CUDA support independently of whether or not Ollama is using GPU. I would also like this to be explained. :)
